avtools/avconv.c

```diff
@@ -1186,13 +1263,21 @@ static void do_streamcopy(InputStream *ist, OutputStream *ost, const AVPacket *p
             opkt.buf = av_buffer_create(opkt.data, opkt.size, av_buffer_default_free, NULL, 0);
             if (!opkt.buf)
                 exit_program(1);
+        } else if (pkt->data != NULL) {
+            opkt.buf = av_buffer_alloc(pkt->size);
```

Comment:
probably just increase the reference count of the packet `buf` would work as well, you probably need to do the same on line 1273